patch/core/book.js

```diff
@@ -52,8 +50,20 @@ module.exports = {
 				const maxlength = a.text.length;
 
 				// make sure new text does not exceed the text it is replacing
-				while (clues[index].length > maxlength) {
-					clues[index] = clues[index].slice(0, maxlength - 1);
+				if (clues[index].length > maxlength) {
+					//trim some words out to try to make it fit
+					clues[index] = clues[index].replace(/ ?mansion/i, '');
+					clues[index] = clues[index].replace(/ ?graveyard/i, '');
+					clues[index] = clues[index].replace(/ ?woods/i, '');
+					clues[index] = clues[index].replace(/ ?cemetery/i, '');
```

Comment:
I like these edits, but I think "cemetery" could cause confusing messages. Changing "Camilla cemetery" to "Camilla" would inevitably lead many players to think that Camilla is holding the item, not that they need garlic to get an item in Camilla cemetery. I would suggest just some custom handling of that one condition, probably up in the `location.replace()` normalization at the top of the file to handle "Camilla cemetery" and just replace it with "cemetery". That, combined with giving garlic secret merchants their own text (as mentioned in a prior review comment) should make the clue much more clear.

Reply:
Also, that series of regexps (after excluding "cemetery" with my aforementioned suggestion) could be streamlined as:

```
clues[index] = clues[index].replace(/ ?(?:mansion|graveyard|woods)/i, '');
```

`(?:)` indicates a "non-capturing group", so it allows you to match on multiple values with getting into all the subgroup matching regexps can do.

Reply:
using the replace anyway it's easy enough to put 
clues[index] = clues[index].replace(/camilla cemetery/i, 'cemetery');
I don't see an advantage to putting them all in one line, but doesn't really matter. If thats how you want it.